connector/connector.go

```diff
@@ -97,3 +97,8 @@ type RefreshConnector interface {
 	// changes since the token was last refreshed.
 	Refresh(ctx context.Context, s Scopes, identity Identity) (Identity, error)
 }
+
+// UserInfoConnector represents connectors that support the user info endpoint
+type UserInfoConnector interface {
+	GetUserInfo(connData []byte, user *map[string]interface{}) (err error)
```

Comment:
[nit] let's not name the return parameter here. While there's no consistent style in the other interfaces above, if all we return is `error`, there's nothing this helps explaining. 😉 What I mean is that above, [for example in HandlePOST](https://github.com/dexidp/dex/pull/1315/files#diff-dac29093b414e413c0bbc846e11d8101R90), there's _two `string` parameters_, so giving them names is illuminating. For one error, it's not. Likewise for [Refresh](https://github.com/dexidp/dex/pull/1315/files#diff-dac29093b414e413c0bbc846e11d8101R98).)